DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/policer/test_policer_basic_functionality.py

```diff
@@ -0,0 +1,166 @@
+    }])
+    assert out[0][dent]['rc'] == 0, 'Failed creating bridge.'
+
+    await IpLink.set(input_data=[{dent: [{'device': bridge, 'operstate': 'up'}]}])
+    assert out[0][dent]['rc'] == 0, 'Failed setting bridge to state UP.'
+
+    # 2. Set link up on interfaces on all participant ports. Enslave all participant ports to the bridge.
+    out = await IpLink.set(input_data=[{dent: [{
+        'device': port,
+        'operstate': 'up',
+        'master': bridge
+    } for port in ports]}])
+    assert out[0][dent]['rc'] == 0, 'Failed setting link to state UP.'
+
+    # 3. Create an ingress qdisc on first port connected to Ixia port/shared block and add a rule with random selectors
+    config = [{'dev': port, 'ingress_block': block, 'direction': 'ingress'} for port in port_with_rule]
+    if qdisc_type == 'port':
+        for item in config:
+            del item['ingress_block']
+
+    out = await TcQdisc.add(input_data=[{dent: config}])
+
+    rule_config = {
+        'action': {'police': {'rate': frame_rate, 'burst': frame_rate + 1000, 'conform-exceed': 'drop'}},
+        'want_ip': False if traffic_type == 'l2' else True,
+        'want_port': True if traffic_type in ['tcp', 'udp'] else None,
+        'skip_sw': True,
+        'want_mac': True if traffic_type == 'l2' else False,
+        'want_tcp': True if traffic_type == 'tcp' else False
+    }
+
+    tc_rule = tcutil_generate_rule_with_random_selectors(port_with_rule, **rule_config)
+
+    # Update tc rule protocol based on traffic type
+    if traffic_type == 'udp':
+        tc_rule['ip_proto'] = 'udp'
+    if traffic_type == 'ipv4':
+        tc_rule['protocol'] = 'ipv4'
+    if traffic_type == 'l2':
+        tc_rule['protocol'] = '0x8100'
+
+    if qdisc_type == 'port':
+        for port in port_with_rule:
+            tc_rule['dev'] = port
+            out = await TcFilter.add(input_data=[{dent: [tc_rule]}])
+            assert out[0][dent]['rc'] == 0, 'Failed to create tc rule'
+    else:
+        del tc_rule['dev']
+        tc_rule['block'] = block
+        out = await TcFilter.add(input_data=[{dent: [tc_rule]}])
+        assert out[0][dent]['rc'] == 0, 'Failed to create tc rule'
+
+    created_rules = {
+        'dev': port_with_rule[0],
+        'direction': 'ingress',
+        'options': '-j'}
+    if qdisc_type == 'shared_block':
+        del created_rules['dev']
+        created_rules['block'] = block
+
+    out = await TcFilter.show(input_data=[{dent: [created_rules]}], parse_output=True)
+    assert out[0][dent]['rc'] == 0, 'Failed to get tc rule'
+
+    # 4 Prepare a matching stream for the randomly selected selectors and with random generated size of packet.
+
+    streams = {}
+    for port in port_with_rule:
+        out = await TcFilter.show(input_data=[{dent: [created_rules]}], parse_output=True)
+        streams.update(tcutil_tc_rules_to_tgen_streams(
+            {port: out[0][dent]['parsed_output']}, frame_rate_type='line_rate', frame_rate_pps=100))
```

Comment:
you can pass multiple rules with multiple ports to `rules_to_tgen` and it will return you a stream for each port for each rule:
```python
{port1: [rule1, rule2, ...],
 port2: [rule3, rule4, ...],
 ...}
```